phoenix-core/src/it/java/org/apache/phoenix/mapreduce/VerifyReplicationToolIT.java

```diff
@@ -0,0 +1,284 @@
+    private static final String SELECT_USER =
+            "SELECT * FROM %s WHERE TENANT_ID = ? LIMIT %d";
+    private static final Random RANDOM = new Random();
+    private static final int NUM_USERS = 10;
+    private static final int NUM_TENANTS = 2;
+
+    private static int tenantNum = 0;
+    private static int userNum = 0;
+    /** source table with 4 regions*/
+    private static String sourceTableName;
+    /** target table with 2 regions*/
+    private static String targetTableName;
+    private List<String> sourceTenants;
+    private String sourceOnlyTenant;
+    private String sourceAndTargetTenant;
+    private String targetOnlyTenant;
+    private PreparedStatement sourceStmt;
+    private PreparedStatement targetStmt;
+    private List<byte[]> sourceSplitPoints;
+    private List<byte[]> targetSplitPoints;
+
+    @BeforeClass
+    public static void createTables() throws Exception {
+        NUM_SLAVES_BASE = 4;
+        Map<String,String> props = Maps.newHashMapWithExpectedSize(1);
+        setUpTestDriver(new ReadOnlyProps(props.entrySet().iterator()));
+        Connection conn = DriverManager.getConnection(getUrl());
+        sourceTableName = generateUniqueName();
+        targetTableName = generateUniqueName();
+        // tables will have the same schema, but a different number of regions
+        conn.createStatement().execute(String.format(CREATE_USER_TABLE, sourceTableName));
+        conn.createStatement().execute(String.format(CREATE_USER_TABLE, targetTableName));
+        conn.commit();
+    }
+
+    @Before
+    public void setupTenants() throws Exception {
+        sourceTenants = new ArrayList<>(NUM_TENANTS);
+        sourceTenants.add("tenant" + tenantNum++);
+        sourceTenants.add("tenant" + tenantNum++);
+        sourceOnlyTenant = sourceTenants.get(0);
+        sourceAndTargetTenant = sourceTenants.get(1);
+        targetOnlyTenant = "tenant" + tenantNum++;
+        /* upsert data and spilt table into multiple regions
+        with each relevant region based on splitpoints
+        residing on a different RS*/
+        upsertData();
+        /* splitting source table into 4 regions */
+        splitSource(sourceTableName);
+        /* splitting target table into 2 regions */
+        splitTarget(targetTableName);
+    }
+
+    @Test
+    public void testVerifyRowsMatch() throws Exception {
+        verify(String.format("TENANT_ID = '%s'", sourceAndTargetTenant), 0, 10, 0, 0, 0, 0);
+    }
+
+    @Test
+    public void testVerifySourceOnly() throws Exception {
+        verify(String.format("TENANT_ID = '%s'", sourceOnlyTenant), 0, 0, 10, 10, 0, 0);
+    }
+
+    @Test
+    public void testVerifyRowsDifferent() throws Exception {
+        // change three rows on the source table so they no longer match on the target
+        upsertSelectData(sourceTableName, sourceAndTargetTenant, -1, 3);
+        verify(String.format("TENANT_ID = '%s'", sourceAndTargetTenant), 0, 7, 3, 0, 0, 3);
+    }
+
```

Comment:
Could we also have a test that verifies that it does the right thing if there are both source-only and target-only rows? I did a whiteboard exercise to prove to myself that it does, but would be good to have it in code too.

Reply:
@gjacoby126 Sorry Wanted to clarify the test scenario, so there should be no common rows between the source and target and what input condition should I verify upon?
Or the source table should have both source-only and target-only rows? A little confused, didn't realize it earlier.

Reply:
@akshita-malhotra - setup is we have Cluster A and Cluster B:
1. Some rows exist only in cluster A
2. Some rows exist in both A and B
3. Some rows exist only in B. 

Assert that your counters are correct. It's essentially combining two of your existing test cases to make sure that it still works with both cases present.

Reply:
Thanks for the clarification @gjacoby126. I updated the PR with the test and the nit. 
There is some refactored code that I need to periodically rebase, I will check if that's needed and update the patch on jira.